plugins/modules/cloud/lxd/lxd_container.py

```diff
@@ -384,6 +411,19 @@ def __init__(self, module):
         self.addresses = None
         self.target = self.module.params['target']
 
+        self.type = self.module.params['type']
+
+        # LXD Rest API provides additional endpoints for creating containers and virtual-machines.
+        self.api_endpoint = None
+        if self.type == 'container' or self.type is None:
+            self.api_endpoint = '/1.0/containers'
+        elif self.type == 'virtual-machine':
+            self.api_endpoint = '/1.0/virtual-machines'
+        else:
+            self.module.fail_json(msg="Validation error: "
+                                      "'type' parameter must be 'container' or 'virtual-machine'. "
+                                      "The current value is '{0}'".format(self.type))
```

Comment:
```suggestion
```
With `choices` Ansible does this validation for you.

Reply:
perfect, less code with the same functionality.